test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -67,8 +67,20 @@ json_spirit::mValue BlockchainTestSuite::doTests(json_spirit::mValue const& _inp
         json_spirit::mObject const& inputTest = i.second.get_obj();
 
         //Select test by name if --singletest is set and not filling state tests as blockchain
-        if (!Options::get().fillchain && !TestOutputHelper::get().checkTest(testname))
-            continue;
+        if (!Options::get().fillchain)
+        {
+            if (test::Options::get().singleTest)
+                BOOST_REQUIRE_MESSAGE(!test::Options::get().singleTestNet.empty(),
+                    "Selecting a single Blockchain test requires --singlenet argument!");
```

Comment:
Maybe you could instead of this check if `test::Options::get().singleTestName` equals beginning of the  `testname` and separately if `test::Options::get().singleTestNet` equals the end of `testname`, then two options would work independently I guess